tests/purs/failing/RecordLabelOnTypeErrorImediate.purs

```diff
@@ -0,0 +1,5 @@
+-- @shouldFailWith TypesDoNotUnify
+module NestedRecordLabelOnTypeError where
+
+record :: { a :: Int }
+record = { a: "a" } -- this should trigger an error, telling us there's a mismatch in the field `a`, but it doesn't
```

Comment:
It's reasonable to leave this test in place, to document that we don't expect a hint in this case, but could you please:
* Amend this comment to reflect that
* Fix the spelling error in the file names (Imediate -> Immediate)